### PR TITLE
Add EnumType by default for PHP >= 8.1

### LINKED ISSUE #281: Add `EnumType` by default for PHP 8.1+
I think we should add the `EnumType` by default for PHP >=8.1

_Originally posted by @roxblnfk in https://github.com/temporalio/sdk-php/issues/278#issuecomment-1411790904_

### PR DESCRIPTION
**Pull request by zlodes**
Closes #281

### CODE REVIEW

File: src/Internal/Marshaller/TypeFactory.php [hunk 1]
```diff
@@ -138,5 +139,9 @@ private function getDefaultMatchers(): iterable
         yield DateIntervalType::class;
         yield ArrayType::class;
         yield ObjectType::class;
+
+        if (PHP_VERSION_ID >= 80104) {
+            yield EnumType::class;
+        }
```
Comment (roxblnfk) [inline src/Internal/Marshaller/TypeFactory.php:145]: It preffer to place it before the `ObjectType`.
Please move the block to the top

Comment (roxblnfk) [reply src/Internal/Marshaller/TypeFactory.php:145]: It need to cover by a test also

Comment (roxblnfk) [reply src/Internal/Marshaller/TypeFactory.php:145]: I'll do it.
I've detected some bugs in EnumType and i'll fix it.

**Comment by roxblnfk** [comment]
Fixed there https://github.com/temporalio/sdk-php/pull/284

@zlodes hi! What about testing for enums without attributes in your project? :)

**Comment by zlodes** [comment]
@roxblnfk hi! Thanks!

Now it works great w/o `#Marshal` attributes! 🚀

Marshaller class seems to be a service (singleton). 
I'll think about an ability to extend available types with custom ones (like `DecimalType` from my example`).

**Comment by roxblnfk** [comment]
Thanks.
Let me know if there is a legal way to add custom type.
If it is impossible then we will improve that.

**Pull request closed without merging**